### DISCUSSION TITLE
TEDAPI Vitals: Not Connected

### DISCUSSION THREAD

**Discussion by timcreech** [jasonacox/Powerwall-Dashboard#558] [Q&A]
I installed the dashboard and everyting went well.  Now I am looking to add the Vitals to get the String data.

I created a route to the GW on my Router.  I can ping 192.168.1.1 from my Rasbperry Pi where this is installed.  I can log onto the Gateway app from my Android phone by going to the 192.16.191.1 address.  I use my Email and a password that I changed previously.

verify.sh sees 192.168.91.1 but Tedapi will not connect

Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): GOOD
 - Version: 0.12.0 Proxy t66
 - Powerwall State: CONNECTED - Firmware: 24.44.0
 - Site Name: Creech Home Energy Gateway
 - Gateway TEDAPI: Available (192.168.91.1)
 - TEDAPI Vitals: Not Connected
 - Cloud Mode: NO

Any thoughts on next area to check?

Thanks!!!

**Comment by jasonacox** [comment]
First, verify that you set the `PW_GW_PWD` variable to the gateway password (usually 10 letter, all caps) in the pypowerwall.env file:

```
PW_EMAIL=youremail@address.com
PW_PASSWORD=
PW_HOST=192.168.91.1
PW_TIMEZONE=America/Los_Angeles
TZ=America/Los_Angeles
PW_DEBUG=no
PW_STYLE=grafana-dark
PW_GW_PWD=ABCDEFGXYZ
```

Next, stop and restart pypowerwall:

```bash
# Stop and remove pypowerwall
docker stop pypowerwall
docker rm pypowerwall

# Restart
./compose-dash.sh up -d

# View Logs
docker logs pypowerwall -f
```

**Reply by timcreech** [reply]
I do have the PW_GW_PWD line in the pypowerwall.env.  But, I am seeing an Access Denied in the logs.

12/18/2024 04:23:38 AM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password
12/18/2024 04:23:38 AM [pypowerwall.tedapi] [ERROR] Failed to connect to Powerwall Gateway
12/18/2024 04:23:39 AM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password

I am using the password that I use when signing into the 192.168.91.1 gateway site.  I changed this password previously when I was doing some other experimenting with the gateway.  

Is there anohter password that I am not thinking of?  Serial Number?

**Reply by jasonacox** [reply]
> I am using the password that I use when signing into the 192.168.91.1 gateway site.

I know this is confusing, but the PW_GW_PWD is the WiFi access point password usually listed on the QR code on the Powerwall to allow installers to commission the system (same one to use when joining the TEG-xxx WiFi access point). It is usually 10 digits and all caps.  It is NOT the same as the password you would use to log in to the web portal.

**Reply by timcreech** [reply]
hmmm, I have tried that one also.  I am able to connect to the TEG-GE0 wifi with the wifi passrod from the Gateway QR code.  Mine is 15 charachters. STG#########GE0.  I have turned debug on and see it is trying to use the PW_GW_PWD that I entered.

I am still thinking of the wrong password?  I know you mentioned Powerwall and I am using the Gateway.  Am I confused?

Log Extract below...

12/19/2024 07:03:57 AM [proxy] [INFO] pyPowerwall [0.12.0] Proxy Server [t66] - HTTP Port 8675 - DEBUG
12/19/2024 07:03:57 AM [pypowerwall] [DEBUG] pypowerwall [0.12.0]

12/19/2024 07:03:57 AM [proxy] [INFO] pyPowerwall Proxy Started
12/19/2024 07:03:57 AM [pypowerwall] [DEBUG] Auto selecting local mode
12/19/2024 07:03:57 AM [pypowerwall] [DEBUG] password = XXXXXX####, gw_pwd = STG#########GE0
12/19/2024 07:03:57 AM [pypowerwall.local.pypowerwall_local] [DEBUG] Tesla local mode enabled
12/19/2024 07:03:57 AM [pypowerwall.local.pypowerwall_local] [DEBUG] loaded auth from cache file .auth/.powerwall (cookie authmode)
12/19/2024 07:03:57 AM [pypowerwall.tedapi] [DEBUG] Testing Connection to Powerwall Gateway: 192.168.91.1
12/19/2024 07:03:57 AM [pypowerwall.tedapi] [DEBUG] Fetching DIN from Powerwall...
12/19/2024 07:03:57 AM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password
12/19/2024 07:03:57 AM [pypowerwall.tedapi] [ERROR] Failed to connect to Powerwall Gateway
12/19/2024 07:03:57 AM [pypowerwall.tedapi] [DEBUG] Testing Connection to Powerwall Gateway: 192.168.91.1
12/19/2024 07:03:57 AM [pypowerwall.tedapi] [DEBUG] Fetching DIN from Powerwall...
12/19/2024 07:03:58 AM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password
12/19/2024 07:03:58 AM [pypowerwall.local.pypowerwall_local] [DEBUG] TEDAPI connection failed - continuing

**Reply by timcreech** [reply]
pypowerwall.env for reference

PW_EMAIL=xx.xxxx@gmail.com
PW_PASSWORD=xxxxxx####
PW_HOST=192.168.91.1
PW_TIMEZONE=America/Los_Angeles
TZ=America/Los_Angeles
PW_DEBUG=no
PW_STYLE=grafana-dark
PW_GW_PWD=STG#########GE0

**Reply by jasonacox** [reply]
If you are using the same password you would use to connect to that WiFi, I don't know why it is not working. Try using the pypowerwall test to ensure it can connect via that password:

```bash
pip install pypowerwall # unless you have it already
python3 -m pypowerwall.tedapi
```

It will prompt for the password. Try your STG#########GE0 and see if that works. If it does, then something is corrupt in the pypowerwall.env file (perhaps some hidden characters or a LF/CR issue if you are using windows).  You can renaming the file and running setup.sh again to have it recreate the file.

**Reply by timcreech** [reply]
Great idea.  I installed and ran and am still getting the Access Denied.....  :( (yes, I did use the actual numnbers and not the # signs)

I connect via wifi with the same pwd to the gateway.  if I run  curl -v -k https://192.168.91.1/api/status, I see the Serial number returned and it matches (minus the leading S)

**CURL output**

{"din":"1118431-00-L--TG#########GE0","start_time":"2024-12-13 04:18:48 +0800","up_time_seconds":"171h40m55.679414693s","is_new":false,"version":"24.44.0 510adfd5","git_hash":"510adfd585b8bd2d897b59af48d0f08e024c1aec","commission_count":0,"device_type":"hec","teg_type":"unknown","sync_type":"v1","cellular_disabled":false,"can_reboot":true}

**Here is the output of python3 -m pypowerwall.tedapi**

pyPowerwall - Powerwall Gateway TEDAPI Reader [v0.12.1]
 - Connecting to https://192.168.91.1... SUCCESS

Enter Powerwall Gateway Password: STG#########GE0

Connecting to Powerwall Gateway 192.168.91.1
Access Denied: Check your Gateway Password
Failed to connect to Powerwall Gateway

ERROR: Unable to connect to Powerwall Gateway. Check your password and try again


Not sure what to try next.

**Reply by timcreech** [reply]
tried about a dozen combinations of the Serial number with S and without S infront, last 5 with and with out S, first 10, last 10, ...

UGH......  can't figure this one out....

**Reply by lignumaqua** [reply]
You don't say but do you happen to have PW3 rather than an earlier model? If so it is the PW3 password you need, not the Gateway one. This password is printed on the label under the PW3 glass cover, visible during installation. If you have multiple PW3 then it's the password from the primary PW3.

**Reply by timcreech** [reply]
Good thought but no, I have a gateway 1 with powerwall 2s

**Reply by jasonacox** [reply]
My Powerwall+ system with a Gateway 2 has two access points and two different QR codes with password. I know, confusing. But only one of them is the right one for TEDAPI. Do you see multiple TEG-xxx WiFi access points? If so, see if you can find the other QR code that may have the right password.  Because you have a Gateway 1, it could be a completely different combination. If you or anyone else with a Gateway 1 is able to get this to work, let us know so we can add to the instructions and troubleshooting helps.

**Reply by timcreech** [reply]
I have tried a few more things.

1) Changed the gateway password bak to the full Serial Number - No Luck
2) Changed the gateway password bak to the last 5 of thel Serial Number - No Luck
3) Changed the gateway password to at least 8 characters, including upper case, lower case, numbers and special characters - No Luck

It seems like there is a separate password for the tedapi access on my Gateway 1

**Reply by jasonacox** [reply]
Ah!  I missed that this is a Gateway 1. We have an issue open on that: https://github.com/jasonacox/Powerwall-Dashboard/issues/536 - It seems that to make it work, you have to toggle the power switch, but that makes it fairly unusable for regular polling of TEDAPI.  You can still use the standard local LAN access but via email/password credentials you would use to log in to the gateway itself, but would not get the extended metrics.

I should add anote about the Gateway 1 limitations in the README/setup. If anyone finds a way to make the Gateway 1 work with the 192.168.91.1 TEDPAI access point, please let us know.
